arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -1040,20 +1079,24 @@ size_t IResearchViewExecutor<ordered, materializeType>::skipAll() {
 
   size_t skipped = 0;
 
-  for (size_t count = this->_reader->size(); _readerOffset < count;) {
-    if (!_itr && !resetIterator()) {
-      continue;
-    }
-
-    while (_itr->next()) {
-      skipped++;
+  if (_readerOffset < this->_reader->size()) {
+    if (this->infos().filterConditionIsEmpty()) {
+      skipped =  this->_reader->live_docs_count();
+      TRI_ASSERT(_totalPos <= skipped);
+      skipped -= std::min(skipped, _totalPos);
+      _readerOffset = this->_reader->size();
+    } else {
+      for (size_t count = this->_reader->size(); _readerOffset < count;
+           ++_readerOffset, _currentSegmentPos = 0) {
+        if (!_itr && !resetIterator()) {
+          continue;
+        }
+        skipped += calculateSkipAllCount(this->infos().countApproximate(), _currentSegmentPos, _itr.get());
+        _itr.reset();
```

Comment:
should `_currentSegmentPos` be set to 0 here?

Reply:
it is done at line 1091